x-pack/plugins/observability_solution/apm/server/routes/assistant_functions/get_log_categories/index.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import datemath from '@elastic/datemath';
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import {
+  SERVICE_NAME,
+  CONTAINER_ID,
+  HOST_NAME,
+} from '../../../../common/es_fields/apm';
+import { getTypedSearch } from '../../diagnostics/create_typed_es_client';
+
+export type LogCategories =
+  | Array<{
+      key: string;
+      docCount: number;
+    }>
+  | undefined;
+
+export async function getLogCategories({
+  esClient,
+  arguments: args,
+}: {
+  esClient: ElasticsearchClient;
+  arguments: {
+    start: string;
+    end: string;
+    'service.name'?: string;
+    'host.name'?: string;
+    'container.id'?: string;
+  };
+}): Promise<LogCategories> {
+  const start = datemath.parse(args.start)?.valueOf()!;
+  const end = datemath.parse(args.end)?.valueOf()!;
+
+  const keyValueFilters = getShouldMatchOrNotExistFilter([
+    { field: SERVICE_NAME, value: args['service.name'] },
+    { field: CONTAINER_ID, value: args['container.id'] },
+    { field: HOST_NAME, value: args['host.name'] },
+  ]);
+
+  const search = getTypedSearch(esClient);
+  const res = await search({
+    index: '*:logs-*,logs-*,*:filebeat*,filebeat*', // TODO: make this configurable somehow (UI, kibana config?)
+    size: 0,
+    track_total_hits: 0,
+    timeout: '5s',
+    query: {
+      bool: {
+        filter: [
+          ...keyValueFilters,
+          { exists: { field: 'message' } },
+          {
+            range: {
+              '@timestamp': {
+                gte: start,
+                lte: end,
+              },
+            },
+          },
+        ],
+      },
+    },
+    aggs: {
+      categories: {
```

Comment:
yes that's fine! I think what you want is something like: we'll not aggregate over more than 500k documents because we know that anything above that is going to be slow.